### PR TITLE
Migrates more ToXContentClasses

### PR DESCRIPTION
**Pull request by colings86**
This includes removing `ToXContentToBytes`

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/index/query/InnerHitBuilder.java [hunk 1]
```diff
@@ -47,7 +47,7 @@
 
 import static org.elasticsearch.common.xcontent.XContentParser.Token.END_OBJECT;
 
-public final class InnerHitBuilder extends ToXContentToBytes implements Writeable {
+public final class InnerHitBuilder implements Writeable, ToXContentObject {
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/index/query/InnerHitBuilder.java]: should we add toString impl to this class now that it doesn't inherit anymore from ToXContentToBytes?

File: core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java [hunk 2]
```diff
@@ -1486,4 +1487,23 @@ public boolean equals(Object obj) {
...
+            if (params.paramAsBoolean("pretty", true)) {
+                builder.prettyPrint();
+            }
+            toXContent(builder, params);
+            return builder.string();
+        } catch (Exception e) {
+            // So we have a stack trace logged somewhere
+            return "{ \"error\" : \"" + ExceptionsHelper.detailedMessage(e) + "\"}";
+        }
+    }
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java]: can we use some of the XContentHelper or Strings.toString here instead?

File: core/src/main/java/org/elasticsearch/index/query/InnerHitBuilder.java (hunk 1 above)
Comment (javanna) [inline core/src/main/java/org/elasticsearch/index/query/InnerHitBuilder.java]: this applies to all the classes that don't extend from ToXCOntentToBytes anymore and don't declare their own toString .

**Review by javanna — CHANGES_REQUESTED** [review]
left a couple of comments, thanks for doing this!

**Comment by colings86** [comment]
@javanna thanks for reviewing, I push two commits that address your comments, could you look again?

File: core/src/main/java/org/elasticsearch/action/admin/indices/recovery/RecoveryResponse.java [hunk 3]
```diff
@@ -127,4 +129,13 @@ public void readFrom(StreamInput in) throws IOException {
             shardRecoveryStates.put(s, list);
         }
     }
+
+    @Override
+    public String toString() {
+        try {
+            return XContentHelper.toXContent(this, XContentType.JSON, true).utf8ToString();
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/action/admin/indices/recovery/RecoveryResponse.java]: Strings.toString already catches IOException. there is a variant of it where you can pass in pretty and himanReadable flags.

**Review by javanna — APPROVED** [review]
LGTM besides the comment I left on using Strings.toString .

**Comment by colings86** [comment]
Note that this PR is only in merged to master and 6.x as the changes are not compatible with 6.0

**Pull request merged**